src/Compilers/CSharp/Portable/Symbols/Source/SourceComplexParameterSymbol.cs

```diff
@@ -1048,6 +1057,199 @@ bool needsReporting()
+            if (constructorArgument.Kind == TypedConstantKind.Primitive)
+            {
+                if (decodeName(constructorArgument, ref arguments) is (int ordinal, var parameter))
+                {
+                    parameterOrdinals = ImmutableArray.Create(ordinal);
+                    parameters = ArrayBuilder<ParameterSymbol?>.GetInstance(1);
+                    parameters.Add(parameter);
+                }
+                else
+                {
+                    // If an error needs to be reported, it will already have been reported by another step.
+                    return;
+                }
+            }
+            else if (constructorArgument.Kind == TypedConstantKind.Array)
+            {
+                bool hadError = false;
+                parameters = ArrayBuilder<ParameterSymbol?>.GetInstance(constructorArgument.Values.Length);
+                var ordinalsBuilder = ArrayBuilder<int>.GetInstance(constructorArgument.Values.Length);
+                foreach (var nestedArgument in constructorArgument.Values)
+                {
+                    if (decodeName(nestedArgument, ref arguments) is (int ordinal, var parameter) && !hadError)
+                    {
+                        parameters.Add(parameter);
+                        ordinalsBuilder.Add(ordinal);
+                    }
+                    else
+                    {
+                        hadError = true;
+                    }
+                }
+
+                if (hadError)
+                {
+                    parameters.Free();
+                    ordinalsBuilder.Free();
+                    return;
+                }
+
+                parameterOrdinals = ordinalsBuilder.ToImmutableAndFree();
+            }
+            else
+            {
+                // There was some error already reported for having an incorrect argument.
+                return;
+            }
+
+            // Validate that the type has a constructor with the correct types.
+            var analyzedArguments = AnalyzedArguments.GetInstance();
+            var intType = this.DeclaringCompilation.GetSpecialType(SpecialType.System_Int32);
+            var intPlaceholder = new BoundInterpolatedStringArgumentPlaceholder(arguments.AttributeSyntaxOpt!, intType, hasErrors: intType.IsErrorType());
+            // literalLength
+            analyzedArguments.Arguments.Add(intPlaceholder);
+            analyzedArguments.RefKinds.Add(RefKind.None);
+            // formattedCount
+            analyzedArguments.Arguments.Add(intPlaceholder);
+            analyzedArguments.RefKinds.Add(RefKind.None);
+
+            foreach (var parameter in parameters)
+            {
+                var argumentType = parameter?.Type ?? ContainingType;
+                analyzedArguments.Arguments.Add(new BoundInterpolatedStringArgumentPlaceholder(arguments.AttributeSyntaxOpt!, argumentType, hasErrors: argumentType.IsErrorType()));
```

Comment:
>BoundInterpolatedStringArgumentPlaceholder [](http://example.com/codeflow?start=52&length=42)

I am curious how ```BoundInterpolatedStringArgumentPlaceholder``` behaves as a target for a ref/out parameter. It is not a variable by default. Are we testing this?

Reply:
We have tests for this side in this PR, but we're also ignoring any diagnostics that come up from this. Regardless, since I'm removing this code it'll be tested in more depth in the next PR.